packages/component/src/Icon/ComponentIcon.tsx

```diff
@@ -20,11 +24,23 @@ const componentIconPropsSchema = pipe(
 type ComponentIconProps = InferInput<typeof componentIconPropsSchema>;
```

Comment:
Maybe need something like below to improve type inference.

```suggestion
type ComponentIconProps = 
  InferInput<typeof componentIconPropsSchema> & 
  {
    icon: // something from CSS modules?
  };
```